src/main/java/net/adoptium/GreetingResource.java

```diff
@@ -1,16 +1,35 @@
 package net.adoptium;
 
 import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
+import java.util.Locale;
+import java.util.MissingResourceException;
+import java.util.ResourceBundle;
 
 @Path("/hello")
 public class GreetingResource {
 
     @GET
     @Produces(MediaType.TEXT_PLAIN)
-    public String hello() {
-        return "Hello World";
+    public String hello(@HeaderParam("Accept-Language") String lang) {
+        System.out.println("------------");
+        System.out.println(lang);
+        ResourceBundle mybundle = null;
+
+        for (Locale.LanguageRange language : Locale.LanguageRange.parse(lang)) {
+            Locale locale = Locale.forLanguageTag(language.getRange());
+            try {
+                mybundle = ResourceBundle.getBundle("MessageBundle", locale);
+                System.out.println("Found language: " + locale);
+                break;
+            } catch (MissingResourceException ignore) {
+                System.out.println("Language not found: " + locale);
```

Comment:
The error must be handled in a better way:
- Error details + stack trace belong to a log
- In case of the error the application should not continue with the regular flow. Throw a RuntimeExeption to stop the current action

Reply:
Also consider whether it makes sense to throw an error or whether there might be a default value that works, too. What do users prefer: error or website in English instead of French?